tests/unit_tests/data_steward/tools/run_deid_test.py

```diff
@@ -16,11 +33,79 @@ def setUpClass(cls):
         print('**************************************************************')
 
     def setUp(self):
-        pass
+        # input parameters expected by the class
+        self.input_dataset = 'foo_input'
+        self.private_key = 'fake/SA/file/path.json'
+        self.output_dataset = 'foo_output_deid'
+        self.action = 'debug'
+        self.skip_tables = 'foo_table'
+        self.tablename = 'bar_table'
+
+        self.correct_parameter_list = [
+            '--idataset', self.input_dataset, '--private_key', self.private_key,
+            '--odataset', self.output_dataset, '--action', self.action,
+            '--skip-tables', self.skip_tables, '--tables', self.tablename
+        ]
+
+        self.incorrect_parameter_list = [
+            '--idataset',
+            self.input_dataset,
+            '--private_key',
+            self.private_key,
+            '--action',
+            self.action,
+            '--skip_tables',
+            self.skip_tables,
+            '--tables',
+            self.tablename,
+        ]
 
     def tearDown(self):
         pass
 
+    def test_parse_args(self):
+        # Tests if incorrect parameters are given
+        self.assertRaises(SystemExit, parse_args, self.incorrect_parameter_list)
+
+        # Preconditions
+        it = iter(self.correct_parameter_list)
+        correct_parameter_dict = dict(zip(it, it))
+        correct_parameter_dict = {
+            k.strip('-').replace('-', '_'): v
+            for (k, v) in correct_parameter_dict.items()
+        }
+
+        # setting correct_parameter_dict values not set in setUp function
+        correct_parameter_dict['console_log'] = False
+        correct_parameter_dict['interactive_mode'] = False
+        correct_parameter_dict['input_dataset'] = self.input_dataset
+
+        # need to delete idataset argument from correct_parameter_dict because input_dataset argument is returned
+        # when self.correct_parameter_list is supplied to parse_args
+        if 'idataset' in correct_parameter_dict:
+            del correct_parameter_dict['idataset']
+
+        # Tests if correct parameters are given
+        results_dict = vars(parse_args(self.correct_parameter_list))
+
+        # Post conditions
+        self.assertEqual(correct_parameter_dict, results_dict)
+
+    @patch('tools.run_deid.fields_for')
+    @patch('tools.run_deid.copy_suppressed_table_schemas')
+    @patch('deid.aou.main')
+    @patch('tools.run_deid.get_output_tables')
+    def test_main(self, mock_tables, mock_main, mock_suppressed, mock_fields):
+        # Tests if incorrect parameters are given
+        self.assertRaises(SystemExit, main, self.incorrect_parameter_list)
+
+        # Preconditions
+        mock_tables.return_value = ['fake1']
+        mock_fields.return_value = {}
+
+        # Tests if correct parameters are given
+        self.assertIsNone(main(self.correct_parameter_list))
+
```

Comment:
We can make this test a bit stronger.  Because you have mocked `aou.main` we need to assert it is called with the parameters we expect.  You can do that with this:

```
        mock_main.assert_called_once_with([
            '--rules', os.path.join(DEID_PATH, 'config', 'ids', 'config.json'),
            '--private_key', self.private_key, '--table', 'fake1',
            '--action', self.action, '--idataset', self.input_dataset,
            '--log', 'LOGS', '--odataset', self.output_dataset])
        self.assertEqual(mock_main.call_count, 1)
```
These two lines will assert the mock is only called once and assert it is called with the expected parameters.

Reply:
This isn't working because you've removed the call to `run_deid.main(self.correct_parameter_list)`.  Because you don't call it, the mock is never used and this post condition isn't met.  I would add

```
<BLANK LINE ON LINE 106>
# run test function
run_deid.main(self.correct_parameter_list)
<BLANK LINE>
# post conditions
mock_main.assert_called_once_with([
    '--rules', os.path.join(DEID_PATH, 'config', 'ids', 'config.json'),
    '--private_key', self.private_key, '--table', 'fake1',
    '--action', self.action, '--idataset', self.input_dataset,
    '--log', 'LOGS', '--odataset', self.output_dataset])
self.assertEqual(mock_main.call_count, 1)
```